Sources/Views/Cells/MediaMessageCell.swift

```diff
@@ -84,14 +84,13 @@ open class MediaMessageCell: MessageContentCell {
     
     /// Handle tap gesture on contentView and its subviews.
     open override func handleTapGesture(_ gesture: UIGestureRecognizer) {
-        let touchLocation = gesture.location(in: self)
-        
-        switch true {
-        case imageView.frame.contains(touchLocation):
-            delegate?.didTapImage(in: self)
-        default:
-            break
+        let touchLocation: CGPoint = convert(gesture.location(in: self), to: imageView)
+
+        guard imageView.bounds.contains(touchLocation) else {
```

Comment:
I don't see the benefit of converting the touch location to imageView and checking bounds over getting the original touch location and checking frame - they produce the same result, and doing the conversion is extra work.

Reply:
Actually, the touchLocation set before is a wrong calculation if you want to compare it with ImageView's frame. Tapping in the middle of the image usually works, but if you tap in the edges, one of them won't work, especially if you have an image with very different width and height.
I had the same problem when I implemented a custom cell to handle link preview, I've just applied the same solution for this case.